src/dashboard/src/main/migrations/0038_fix_workflow_store_dip.py

```diff
@@ -0,0 +1,189 @@
+       SIP is still where MCPServer expects it to be and preconfigured choices
+       can work for Store DIP and Store DIP locations
+    """
+    ###########################################################################
+    # Model Classes
+    ###########################################################################
+
+    MicroServiceChain = apps.get_model(
+        'main', 'MicroServiceChain')
+    MicroServiceChainLink = apps.get_model(
+        'main', 'MicroServiceChainLink')
+    MicroServiceChainChoice = apps.get_model(
+        'main', 'MicroServiceChainChoice')
+    StandardTaskConfig = apps.get_model(
+        'main', 'StandardTaskConfig')
+    TaskConfig = apps.get_model(
+        'main', 'TaskConfig')
+
+    ###########################################################################
+    # Useful Model Instances
+    ###########################################################################
+
+    # The "Reject DIP" chain/link will be destroyed.
+    reject_dip_chain_choice_uuid = '39bcba03-d251-4974-8a7b-45b2444e19a8'
+    reject_dip_link_uuid = 'f2a1faaf-7322-4d9c-aff9-f809e7a6a6a2'
+    reject_dip_chain_uuid = 'eea54915-2a85-49b7-a370-b1a250dd29ce'
+    reject_dip_tc_uuid = 'ea331cfb-d4f2-40c0-98b5-34d21ee6ad3e'
+    reject_dip_stc_uuid = '4f7e2ed6-44b9-49a7-a1b7-bbfe58eadea8'
+
+    # "Store DIP" chain: its description will be changed to "Do Not Upload DIP"
+    # and its starting link will be the "Store DIP?" decision point.
+    store_dip_quest_link_uuid = '5e58066d-e113-4383-b20b-f301ed4d751c'
+
+    # Links to prune (destroy)
+    dstry_retrieve_locat_link_uuid = 'ed5d8475-3793-4fb0-a8df-94bd79b26a4c'
+    dstry_store_dip_locat_link_uuid = 'b7a83da6-ed5a-47f7-a643-1e9f9f46e364'
+    dstry_store_dip_link_uuid = 'e85a01f1-4061-4049-8922-5694b25c23a2'
+    dstry_move_uploaded_link_uuid = 'e3efab02-1860-42dd-a46c-25601251b930'
+
+    # "Completed" link becomes "Handle unstored DIP" and its STC references a
+    # new execute (and arguments), which is a new client script that moves the
+    # DIP to the rejected/ directory iff it has NOT been uploaded.
+    completed_link_uuid = 'f8ee488b-5667-4417-ae15-bed9e42ee97d'
+    completed_stc_uuid = '888281a1-9678-46ed-a1a0-be9f0c6d02b0'
+
+    upload_dip_choice_link_uuid = '92879a29-45bf-4f0b-ac43-e64474f0f2f9'
+
+    ###########################################################################
+    # 1. Remove "Reject DIP" from options at "Upload DIP" decision point:
+    ###########################################################################
+
+    # Destroy the "Reject DIP" chain and link and attendant models.
+    # For some unknown reason, deleting just these two models will result in a
+    # deletion of the three that follow, despite the fact that this does not
+    # seem to be entailed by Django's foreign key cascade delete strategy.
+    StandardTaskConfig.objects.get(id=reject_dip_stc_uuid).delete()
+    TaskConfig.objects.get(id=reject_dip_tc_uuid).delete()
```

Comment:
`TaskConfig` is referenced by `MicroServiceChainLink. currenttask`. `MicroServiceChainLink` is referenced by `MicroServiceChainChoice. choiceavailableatlink` and so on. So I'd assume that the behaviour you're seeing is because of Django is emulating `ON DELETE CASCADE`?